main/FatesHistoryInterfaceMod.F90

```diff
@@ -2206,6 +2208,14 @@ subroutine update_history_dyn(this,nc,nsites,sites)
                     hio_biomass_si_agepft(io_si,iagepft) = hio_biomass_si_agepft(io_si,iagepft) + &
                           total_c * ccohort%n * AREA_INV
 
+                    hio_nplant_si_agepft(io_si,iagepft) = hio_nplant_si_agepft(io_si,iagepft) + &
+                          ccohort%n
+
+                    !!! Scorch Height.  Should be identical for all cohorts of a given PFT in a given patch.  
+                    !!! in the event that there is more than one patch in this age bin, weight the answer by 
+                    !!! the number of individuals in the respective patches.
+                    hio_scorch_height_si_agepft(io_si,iagepft) = hio_scorch_height_si_agepft(io_si,iagepft) + &
+                          ccohort%Scorch_ht * ccohort%n
```

Comment:
yeah the weighting for this is a bit non-intuitive.  i was thinking that the possibilities are to write just the `ccohort%Scorch_ht`, to weight by PFT as I dd, or to weight by patch area.  I didn't want to just make it the last cohort thaqt got written, so one of the latter made more sense.  Since the variable itself is attached to the cohort, it made sense to do the weighting by cohort%n since we encounter the variable within a cohort loop.  The other possibility is to not make `Scorch_ht` a cohort variable at all, and instead make it a patch variable with a numPFT array.  That actually has a memory advantage.  so if we want to be most efficient, we could do that?  I've got all the stuff open right now, shall I make that change?

Reply:
wait, I'm looking at theg code now and am confused.  in calculating scorch_ht, why do the biomass check at all, and  if it is needed, why do it sequentially through cohorts while calculating scorch_ht in the loop?  Isn't it correct to first evaluate if there is some tree AGB on a patch, and then if so, to calculate the scorch_heitght for that PFT in that patch?  otherwise there's an order dependency to it.  probably a trivial one, but one nonetheless.

Reply:
@ckoven I think you are right. I could have just made scorch height the same for all the PFTs in the patch. so you are saying take it out of the cohort loop, and do this as a patch loop.

Reply:
each PFT in the patch has it's own scorch height...

Reply:
I think I left it that way because of the way crown damage is calculated. The scorch height is compared to the cohort height to get canopy burnt..
https://github.com/jkshuman/fates/blob/e5e808ca3f7c4f65e4a5fd3d6dbee61a639ac9ab/fire/SFMainMod.F90#L882-L894

But can look at tree AGB at patch level and then do the cohort loop calculation. Is that what you are suggesting?

Reply:
The change you are suggesting makes sense and simplifies things. Want me to accept your changes and make it on my end?

Reply:
I've got something written and am compiling it right now.  I'll update my PR if it works.